### PR TITLE
Update ORT Training to 1.14.0

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

- [x] Create Dockerfile for onnxruntime-training 1.14.0
- [x] Update trainer documentation
- [x] Complete the fix in #737 (I only did the patch for `evaluation_loop_ort()` but not for `prediction_loop_ort`, but it is the same content)

### CODE REVIEW

File: optimum/onnxruntime/trainer.py [hunk 1]
```diff
@@ -1210,14 +1211,10 @@ def prediction_loop_ort(
...
         args = self.args
-        # Temporary fix for decoder, now `use_cache` set to False which cause overhead on re-computing past keys and values.
-        # TODO: Use cache once `ORTModelForCausalLM` supports `loss` as output
         if ort_model_cls is ORTModelForCausalLM:
-            ort_model = ort_model_cls.from_pretrained(
-                model_id=model_id, use_cache=False, provider="CUDAExecutionProvider"
-            )
+            ort_model = ort_model_cls.from_pretrained(model_id=model_id, use_cache=use_cache).to(args.device)
         else:
-            ort_model = ort_model_cls.from_pretrained(model_id=model_id, provider="CUDAExecutionProvider")
```
Comment (fxmarty) [inline optimum/onnxruntime/trainer.py:1220]: Why this change only in the `else` branch?

**Review by fxmarty — APPROVED** [review]
LGTM, thank you!

File: optimum/onnxruntime/trainer.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/onnxruntime/trainer.py:1220]: you mean `use_cache` or the ep?
EP are changed for both branch, `use_cache` is only needed for decoder cases(if branch) not other cases(else branch).

Comment (fxmarty) [reply optimum/onnxruntime/trainer.py:1220]: I meant `provider="CUDAExecutionProvider"` to `.device(args.device)`

Comment (JingyaHuang) [reply optimum/onnxruntime/trainer.py:1220]: ? From what I see both branches are changes from precising cuda ep to passing to args.device.

https://github.com/huggingface/optimum/blob/e537404799b478cd890d298ad01516db749efcc6/optimum/onnxruntime/trainer.py#L1214-L1217
